src/nupic/encoders/ScalarSensor.cpp

```
@@ -0,0 +1,309 @@
+        0, // elementCount
+        true, // isRegionLevel
+        true // isDefaultOutput
+        ));
+
+    ns->outputs.add(
+      "bucket",
+      OutputSpec(
+        "Bucket number for this sensedValue",
+        NTA_BasicType_Int32,
+        0, // elementCount
+        true, // isRegionLevel
+        false // isDefaultOutput
+        ));
+
+    return ns;
+  }
+
+  void
+  ScalarSensor::getParameterFromBuffer(const std::string& name,
+                                       Int64 index,
+                                       IWriteBuffer& value)
+  {
+    if (name == "sensedValue")
+    {
+      value.write(sensedValue_);
+    }
+    else if (name == "n")
+    {
+      value.write(encoder_->getOutputWidth());
+    }
+    else
+    {
+      NTA_THROW << "ScalarSensor::getParameter -- Unknown parameter " << name;
+    }
+  }
+
+  void
+  ScalarSensor::setParameterFromBuffer(const std::string& name,
+                                       Int64 index,
+                                       IReadBuffer& value)
+  {
+    if (name == "sensedValue")
+    {
+      value.read(sensedValue_);
+    }
+    else
+    {
+      NTA_THROW << "ScalarSensor::setParameter -- Unknown parameter " << name;
+    }
+  }
+
+  void
+  ScalarSensor::initialize()
+  {
+    encodedOutput_ = getOutput("encoded");
+    bucketOutput_ = getOutput("bucket");
+  }
+
+  size_t
+  ScalarSensor::getNodeOutputElementCount(const std::string& outputName)
+  {
+    if (outputName == "encoded")
+    {
+      return encoder_->getOutputWidth();
+    }
+    else if (outputName == "bucket")
+    {
+      return 1;
+    }
+    else
+    {
+      NTA_THROW << "ScalarSensor::getOutputSize -- unknown output " << outputName;
+    }
+  }
+
+  std::string ScalarSensor::executeCommand(const std::vector<std::string>& args, Int64 index)
+  {
+    NTA_THROW << "ScalarSensor::executeCommand -- commands not supported";
+  }
+
+  void ScalarSensor::serialize(BundleIO& bundle)
+  {
+    // TODO
```

Can you raise an exception for now so it doesn't silently fail?